packages/liveblocks-react/src/liveblocks.tsx

```diff
@@ -666,24 +666,29 @@ function useUser_withClient<U extends BaseUserMeta>(
     [usersStore, userId]
   );
 
-  useEffect(() => {
-    // NOTE: .get() will trigger any actual fetches, whereas .getState() will not
-    void usersStore.get(userId);
-  }, [usersStore, userId]);
-
   const selector = useCallback(
     (state: ReturnType<typeof getUserState>) =>
       selectorFor_useUser(state, userId),
     [userId]
   );
 
-  return useSyncExternalStoreWithSelector(
+  const result = useSyncExternalStoreWithSelector(
     usersStore.subscribe,
     getUserState,
     getUserState,
     selector,
     shallow
   );
+
+  // Trigger a fetch if we don't have any data yet (whether initially or after an invalidation)
+  useEffect(() => {
+    if (!result.user && result.isLoading) {
+      // NOTE: .get() will trigger any actual fetches, whereas .getState() will not
+      void usersStore.get(userId);
+    }
+  }, [usersStore, userId, result]);
+
+  return result;
```

Comment:
Not entirely sure why this change is needed? My expectation would be that `void usersStore.get(userId)` would not do anything if it's already loading or successfully loaded. I.e. this condition should get checked by the `usersStore.get()` method, not here, not?

Reply:
The invalidate methods don't trigger a revalidation on their own, it's up to the hooks (or any other consumer) to trigger one (to make sure we only revalidate if needed, e.g. at least one consumer is rendered and needs revalidated data). This change here makes the effect re-trigger if `result` goes back to an initial state, it's not related to deduping but I can see why the `if` statement would make it look that way, I'll remove it because yeah it's technically unnecessary.

Reply:
Improved in https://github.com/liveblocks/liveblocks/pull/1992/commits/25abd0b3eb9d59c81cddade5f48de59522abe419.